gateway-ha/gateway-ha-config.yml

```diff
@@ -36,8 +36,11 @@ monitor:
 
 modules:
   - io.trino.gateway.ha.module.HaGatewayProviderModule
-  - io.trino.gateway.ha.module.ClusterStateListenerModule
+#  - io.trino.gateway.ha.module.ClusterStateListenerModule
+#  - io.trino.gateway.ha.module.BasicRouterProvider
   - io.trino.gateway.ha.module.ClusterStatsMonitorModule
+  - io.trino.gateway.ha.module.QueueLengthListenerModule
+  - io.trino.gateway.ha.module.QueueLengthRouterProvider
```

Comment:
The `QueueLengthRouterProvider` requires that the `QueueLengthListenerModule` is configured (right?). If it isn't, what kind of error does the end user see? 

If its not an easy one to understand, we should add some checking in Baseapp to ensure that dependent modules are loaded. `AppConfiguration.getModules()` should give us the list of loaded modules.

Reply:
In my latest change, I am providing the BasicRouter by default and config can override it. Is that ok?

Reply:
sgtm

Reply:
done.